### PR TITLE
Un-flake test_runtime_configurable_cache_size

### LINKED ISSUE #66887: Flaky `test_runtime_configurable_cache_size`
https://s3.amazonaws.com/clickhouse-test-reports/66850/1be54641c16569e7ee3398fe0770a9b532342643/integration_tests__asan__old_analyzer__[1_6].html

It doesn't fail often but it did fail couple of times.

### PR DESCRIPTION
**Pull request by rschu1ze**
Fixes #66887

This PR simplifies the test such that it can no longer be affected by timing.

For simplicity, the check against `system.asynchronous_metrics` is replaced by a check against `system.query_cache`. A new stateless test to check asynchronous metrics was added instead.

Kudos to @antonio2368: https://github.com/ClickHouse/ClickHouse/issues/66887#issuecomment-2244745430

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: tests/integration/test_runtime_configurable_cache_size/test.py [hunk 1]
```diff
@@ -94,51 +94,49 @@ def start_cluster():
...
+    node.query("SELECT 3 SETTINGS use_query_cache = 1, query_cache_ttl = 1")
     res = node.query(
-        "SELECT value FROM system.asynchronous_metrics WHERE metric = 'QueryCacheEntries'",
+        "SELECT count(*) FROM system.query_cache",
     )
-    assert res == "1\n"
+    assert res == "0\n"
 
     # restore the original config
     node.copy_file_to_container(
```
Comment (antonio2368) [inline tests/integration/test_runtime_configurable_cache_size/test.py:142]: Run a query and verify that the count is 1?

**Review by antonio2368 — APPROVED** [review]

File: tests/integration/test_runtime_configurable_cache_size/test.py (hunk 1 above)
Comment (rschu1ze) [reply tests/integration/test_runtime_configurable_cache_size/test.py:142]: Makes sense, done.

**Review by antonio2368 — APPROVED** [review]

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 919005c4f70b044ecd9cc1bbce5dc5e276e11929 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66934/919005c4f70b044ecd9cc1bbce5dc5e276e11929/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/91693daf599bbaeaed5adb8cba257c176628e2be/docs_check.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/bab5565b654e62481c2c3b4a144ab6f7d5fc92ea/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (aarch64) [3/6]: https://github.com/ClickHouse/ClickHouse/issues/67002
ClickHouse Stateless Tests (tsan, S3 Storage) [4/4]: https://github.com/ClickHouse/ClickHouse/pull/66188

**Pull request merged**